pipeline_dp/dp_engine.py

```diff
@@ -25,10 +27,12 @@ class DPEngine:
     """Performs DP aggregations."""
 
     def __init__(self, budget_accountant: BudgetAccountant,
-                 ops: PipelineOperations):
+                 ops: PipelineOperations, 
+                 partition_selection_strategy: Optional[PartitionSelectionStrategy] = None):
         self._budget_accountant = budget_accountant
         self._ops = ops
         self._report_generators = []
+        self._partition_selection_strategy = partition_selection_strategy
```

Comment:
Partitions selection strategy can be different for different aggregations, so let's create it inside _select_private_partions (more details in comments there)